mitiq/zne/tests/test_inference.py

```diff
@@ -872,6 +872,86 @@ def test_get_methods_of_factories():
     assert np.allclose(fac.get_zero_noise_limit_error(), 1.0)
 
 
+@mark.parametrize(
+    "factory", (LinearFactory, RichardsonFactory, PolyFactory,),
+)
+def test_plot_data(factory):
+    """Test that plot_data() outputs the correct x and y values."""
+    if factory is PolyFactory:
+        fac = factory(scale_factors=X_VALS, order=3)
+    else:
+        fac = factory(scale_factors=X_VALS)
+    fac.run_classical(apply_seed_to_func(f_lin, SEED))
+    fac.reduce()
+    fig = fac.plot_data()
+    ax = fig.gca()
+    x_data, y_data = ax.lines[0].get_xydata().T
+    zne_curve = fac.get_extrapolation_curve()
+    np.testing.assert_equal(
+        np.any(np.not_equal(y_data, zne_curve(x_data))), True,
+    )
```

Comment:
If I understand well, I think that here we want to test that `y_data` is equal to `zne_curve(x_data)` up to numerical rounding errors.

Since testing perfect equality with real elements is problematic, what about replacing the last instruction with:
`assert np.allclose(y_data, zne_curve(x_data))` ?

Side comment: I was also confused by
```
np.testing.assert_equal(
        np.any(np.not_equal(y_data, zne_curve(x_data))), True,
    )
```
since I expected:
```
    zne_curve = fac.get_extrapolation_curve()
    np.testing.assert_equal(
        np.any(np.not_equal(y_data, zne_curve(x_data))), False,
    )
```

But, anyway, using `np.allclose...` is probably optimal.

Reply:
Yeah I agree, I have swapped to using the `np.allclose` as per your suggestion.